src/main/java/de/qabel/desktop/config/RepositoryBasedClientConfig.java

```diff
@@ -0,0 +1,174 @@
+package de.qabel.desktop.config;
+
+import de.qabel.core.config.Account;
+import de.qabel.core.config.Identity;
+import de.qabel.core.drop.DropURL;
+import de.qabel.desktop.repository.*;
+import de.qabel.desktop.repository.exception.EntityNotFoundExcepion;
+import de.qabel.desktop.repository.exception.PersistenceException;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+
+public class RepositoryBasedClientConfig implements ClientConfig {
+    public static final String ACCOUNT_KEY = "account";
+    private final ClientConfigRepository configRepo;
+    private final AccountRepository accountRepo;
+    private final IdentityRepository identityRepo;
+    private final DropStateRepository dropStateRepo;
+    private final ShareNotificationRepository shareRepo;
+    private final List<Consumer<Identity>> identityListener = new CopyOnWriteArrayList<>();
+    private final List<Consumer<Account>> accountListener = new CopyOnWriteArrayList<>();
+
+    public RepositoryBasedClientConfig(
+        ClientConfigRepository configRepo,
+        AccountRepository accountRepo,
+        IdentityRepository identityRepo,
+        DropStateRepository dropStateRepo,
+        ShareNotificationRepository shareRepo
+    ) {
+        this.configRepo = configRepo;
+        this.accountRepo = accountRepo;
+        this.identityRepo = identityRepo;
+        this.dropStateRepo = dropStateRepo;
+        this.shareRepo = shareRepo;
+    }
+
+    @Override
+    public boolean hasAccount() {
+        try {
+            return configRepo.contains(ACCOUNT_KEY);
+        } catch (PersistenceException e) {
+            throw new IllegalStateException("failed to check account", e);
+        }
+    }
+
+    @Override
+    public Account getAccount() {
+        try {
+            return accountRepo.find(configRepo.find("account"));
+        } catch (PersistenceException | EntityNotFoundExcepion e) {
+            throw new IllegalStateException("couldn't load current account", e);
+        }
+    }
+
+    @Override
+    public void setAccount(Account account) throws IllegalStateException {
+        try {
+            accountRepo.save(account);
+            configRepo.save("account", String.valueOf(account.getId()));
+            accountListener.forEach(c -> c.accept(account));
+        } catch (PersistenceException e) {
+            throw new IllegalStateException("failed to set account", e);
+        }
+    }
+
+    @Override
+    public void onSetAccount(Consumer<Account> consumer) {
+        accountListener.add(consumer);
+    }
+
+    @Override
+    public Identity getSelectedIdentity() {
+        try {
+            if (!configRepo.contains("identity") || configRepo.find("identity") == null) {
+                return null;
+            }
+            return identityRepo.find(Integer.valueOf(configRepo.find("identity")));
```

Comment:
concerning the integer:
The configRepo is ment to be a generic key-value store (because the repo is ment to stay as stable as possible and souldn't be client specific but most config values will be). The RepositoryBasedClientConfig then acts as the glue between the configRepo with it's semanticless values and the other repos that need the correct types. Different clients can use differend implementations as glue for their specific needs ... I couldn't come up with a better solutions, ideas welcome...